transformer_engine/pytorch/ops/basic/basic_linear.py

```diff
@@ -389,6 +406,32 @@ def _functional_forward(
                 "are not compatible"
             )
 
+        # Check output tensor dims
```

Comment:
I wonder if we need to do this here (same for input) or maybe we could rely on the error checking on the C++ side to minimize CPU overhead?

Reply:
I think that would be a good optimization in the future, especially since the linear functional API is used in multiple operations.